src/frontend/src/flows/authenticate/index.ts

```diff
@@ -0,0 +1,390 @@
+import { html, render } from "lit-html";
+import { editIcon, undoIcon } from "../../components/icons";
+import { navbar } from "../../components/navbar";
+import { footer } from "../../components/footer";
+import {
+  getUserNumber,
+  parseUserNumber,
+  setUserNumber,
+} from "../../utils/userNumber";
+import { withLoader } from "../../components/loader";
+import { IIConnection } from "../../utils/iiConnection";
+import {
+  apiResultToLoginFlowResult,
+  LoginFlowResult,
+} from "../login/flowResult";
+import { displayError } from "../../components/displayError";
+import { useRecovery } from "../recovery/useRecovery";
+import { initRegistration } from "../../components/registrationLink";
+import waitForAuthRequest, { AuthContext } from "./postMessageInterface";
+import { toggleErrorMessage } from "../../utils/errorHelper";
+import { fetchDelegation } from "./fetchDelegation";
+
+const pageContent = (
+  hostName: string,
+  maxTimeToLive: bigint | undefined,
+  userNumber?: bigint
+) => html` <style>
+    .spacer {
+      height: 2rem;
+    }
+
+    .spacer-small {
+      height: 1rem;
+    }
+
+    .anchorText {
+      font-size: 1.5rem;
+    }
+
+    #userNumberInput {
+      margin: 0;
+      text-align: center;
+      width: 100%;
+      box-sizing: border-box;
+      padding: 0.5rem 2rem;
+    }
+
+    #userNumberInput:focus {
+      box-sizing: border-box;
+      border-style: double;
+      border-width: 2px;
+      border-radius: 4px;
+      border-image-slice: 1;
+      outline: none;
+      border-image-source: linear-gradient(
+        270.05deg,
+        #29abe2 10.78%,
+        #522785 22.2%,
+        #ed1e79 42.46%,
+        #f15a24 59.41%,
+        #fbb03b 77.09%
+      );
+    }
+
+    .modeContainer {
+      min-height: 6rem;
+    }
+
+    .childContainer {
+      position: relative;
+      margin-bottom: 0.5rem;
+    }
+
+    .switchButton {
+      display: flex;
+      flex-direction: column;
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      right: 0;
+      padding: 0.3rem;
+      margin: 0;
+      width: 2rem;
+      background: transparent;
+      border: none;
+    }
+
+    .smallText {
+      font-size: 0.875rem;
+      font-weight: 400;
+    }
+
+    .sectionTitle {
+      margin: 1rem 0 0.5rem 0;
+      color: black;
+    }
+
+    hr {
+      border-color: var(--grey-050);
+      margin: 1rem 0;
+    }
+
+    .not-displayed {
+      display: none;
+    }
+  </style>
+  <div class="container">
+    <h1>Authorize private login</h1>
```

Comment:
"private" as opposed to what?

Reply:
This was suggested by Jan as a hint to the per application principals. I think it's meant as private in the sense of privacy-preserving, as opposed to public / linkable.